### ISSUE TITLE
Videos with a pixel format of yuv444p result in incorrect "Unsupported h264 feature: interlace" exception

### ISSUE THREAD

**Issue by sknick** [jcodec/jcodec#380]
If I attempt to decode frames from a video with a pixel format of yuv444p, an exception is thrown with an incorrect error message indicating that the video has interlacing enabled and that is the issue:

```
Caused by: java.lang.RuntimeException: Unsupported h264 feature: interlace.
	at org.jcodec.codecs.h264.H264Decoder$FrameDecoder.validateSupportedFeatures(H264Decoder.java:217)
	at org.jcodec.codecs.h264.H264Decoder$FrameDecoder.init(H264Decoder.java:190)
	at org.jcodec.codecs.h264.H264Decoder$FrameDecoder.decodeFrame(H264Decoder.java:139)
	at org.jcodec.codecs.h264.H264Decoder.decodeFrameFromNals(H264Decoder.java:103)
	at org.jcodec.codecs.h264.H264Decoder.decodeFrame(H264Decoder.java:99)
	at org.jcodec.api.specific.AVCMP4Adaptor.decodeFrame(AVCMP4Adaptor.java:68)
	at org.jcodec.api.FrameGrab.getNativeFrameWithMetadata(FrameGrab.java:227)
```

As you can see from this mediainfo output, the scan type is Progressive:

```
General
Complete name                            : ./TestVideo.out.mp4
Format                                   : MPEG-4
Format profile                           : Base Media
Codec ID                                 : isom (isom/iso2/avc1/mp41)
File size                                : 65.9 KiB
Duration                                 : 11s 0ms
Overall bit rate                         : 49.0 Kbps
Encoded date                             : UTC 1904-01-01 00:00:00
Tagged date                              : UTC 1904-01-01 00:00:00
Writing application                      : Lavf56.40.101

Video
ID                                       : 1
Format                                   : AVC
Format/Info                              : Advanced Video Codec
Format profile                           : High 4:4:4 Predictive@L3.1
Format settings, CABAC                   : Yes
Format settings, ReFrames                : 4 frames
Codec ID                                 : avc1
Codec ID/Info                            : Advanced Video Coding
Duration                                 : 11s 0ms
Bit rate                                 : 45.6 Kbps
Width                                    : 1 024 pixels
Height                                   : 792 pixels
Display aspect ratio                     : 1.293
Frame rate mode                          : Constant
Frame rate                               : 30.000 fps
Color space                              : YUV
Chroma subsampling                       : 4:4:4
Bit depth                                : 8 bits
Scan type                                : Progressive
Bits/(Pixel*Frame)                       : 0.002
Stream size                              : 61.2 KiB (93%)
Writing library                          : x264 core 148
Encoding settings                        : cabac=1 / ref=3 / deblock=1:0:0 / analyse=0x3:0x113 / me=hex / subme=7 / psy=1 / psy_rd=1.00:0.00 / mixed_ref=1 / me_range=16 / chroma_me=1 / trellis=1 / 8x8dct=1 / cqm=0 / deadzone=21,11 / fast_pskip=1 / chroma_qp_offset=4 / threads=4 / lookahead_threads=1 / sliced_threads=0 / nr=0 / decimate=1 / interlaced=0 / bluray_compat=0 / constrained_intra=0 / bframes=3 / b_pyramid=2 / b_adapt=1 / b_bias=0 / direct=1 / weightb=1 / open_gop=0 / weightp=2 / keyint=250 / keyint_min=25 / scenecut=40 / intra_refresh=0 / rc_lookahead=40 / rc=crf / mbtree=1 / crf=23.0 / qcomp=0.60 / qpmin=0 / qpmax=69 / qpstep=4 / ip_ratio=1.40 / aq=1:1.00
Encoded date                             : UTC 1904-01-01 00:00:00
Tagged date                              : UTC 1904-01-01 00:00:00
```

If I re-encode the video using yuv420p, it decodes fine as you might expect.

**Comment by svitvitskiy** [comment]
Could you please attach your video to this ticket so I could take a look?

**Comment by sknick** [comment]
Unfortunately, it's not a video I can upload. I think you can reproduce pretty easily though by creating a video with a YUV444 profile though.

**Comment by svitvitskiy** [comment]
For now this is intended behavior, won't fix. I added a test case for that in 39d1bda

I used this command to compress a yuv444 video:
ffmpeg -y -s 720x576 -pix_fmt yuv444p -i squirrel-720x576-444P.yuv -c:v libx264 -pix_fmt yuv444p -profile:v high444 -crf 0 -preset:v slow squirrel-720x576-444P.mp4
It did contain an interlacing flag even though the video itself was not interlaced, I guess this might be automatically included because of the 'high444' profile. In any case, we don't support 444 video right now.
